processor/src/main/java/com/github/victornoel/eo/apt/GeneratedEnvelopeTypeSpec.java

```diff
@@ -68,7 +77,12 @@
      */
     public GeneratedEnvelopeTypeSpec(final TypeElement source,
         final ProcessingEnvironment procenv) {
-        this(source, new GeneratedEnvelopeName(source).get(), procenv);
+        this(
+            source,
+            new GeneratedEnvelopeName(source).get(),
+            procenv,
+            new NameAllocator()
```

Comment:
@andreoss could introduce a class named `AllocatedNames` that is responsible of allocating the existing names to the `NameAllocator`? Like this we can properly separate responsibility. A bit like `GeneratedEnvelopeName` above.

Something like `new AllocatedNames(source, procenv).get()` would be perfect

Reply:
@victornoel Done. It also made sense to extract an interface for name generation.